micro_epsilon_scancontrol_driver/src/driver.cpp

```diff
@@ -550,6 +564,100 @@ namespace scancontrol_driver
         // return true;
     }  
 
+    // Generic function for setting a times like exposure and idle time.
+    // setting_id: 
+    // value: time in microseconds. min: 1 mus, max: 40950 mus ;  eg. 1005 = 1.005ms
+    int ScanControlDriver::SetTime(unsigned int setting_id, unsigned int value){
+        int ret_code;
+
+        // encoded value in 1 mus steps. 
+        uint32_t remainder = ((value % 10) << 12) & 0xF000; // Remainder is left shifted first and bits 0-12 are masked. the quotient will occupy this area.
+        uint32_t quotient = ((value / 10)) & 0xFFF; // take the quotient and mask bits 12-15
+        uint32_t value_encoded = remainder + quotient; 
```

Comment:
nitpick: as you use `actual_value` later this is a little more consistent in naming.
```suggestion
        uint32_t encoded_value = remainder + quotient; 
```

Reply:
Changed